src/server/info_collector.cpp

```diff
@@ -150,15 +146,12 @@ void info_collector::on_app_stat()
         // get row data statistics for all of the apps
         all_stats.merge(app_stats);
 
-        // hotspot_calculator is to detect hotspots
-        hotspot_calculator *hotspot_calculator =
+        // hotspot_partition_calculator is to detect hotspots
+        hotspot_partition_calculator *hotspot_partition_calculator =
             get_hotspot_calculator(app_rows.first, app_rows.second.size());
-        if (!hotspot_calculator) {
-            continue;
-        }
-        hotspot_calculator->aggregate(app_rows.second);
-        // new policy can be designed by strategy pattern in hotspot_partition_data.h
-        hotspot_calculator->start_alg();
+        dassert(hotspot_partition_calculator != nullptr, "hotspot_partition_calculator is NULL");
```

Comment:
I suggest printing more information such as `app_name`, `partition_count` in assert.
Besides, is it have to assert here? If the hotspot_partition_calculator is not found, `on_app_stat` can also work.